packages/jazz-tools/src/tools/subscribe/errorReporting.ts

```diff
@@ -0,0 +1,135 @@
+import type { CoValue, ID } from "../internal.js";
+import { CoValueLoadingState } from "./types.js";
+
+/**
+ * Check if we're in development mode.
+ * Stack traces are only captured in development to avoid overhead in production.
+ */
+export function isDev(): boolean {
+  return process.env.NODE_ENV === "development";
+}
+
+/**
+ * Capture a stack trace only in development mode.
+ * Returns empty string in production to avoid overhead.
+ */
+export function captureStack(): string {
+  return isDev() ? new Error().stack || "" : "";
+}
```

Comment:
This would be a perfect use-case for Error.cause. Essentially all JS engines now support it.

```suggestion
export function captureCause() {
  return new Error(`The Jazz error originated here`);
}
```

and then something like this
```ts
console.error(new Error(`Jazz Unavailable Error: ${value.toString()}`, { cause: callerCause }));
```

would look like this on a browser

<img width="700" height="130" alt="Image" src="https://github.com/user-attachments/assets/aed2829c-55fe-4362-8186-ff859ce45900" />